src/commands/openYAMLConfigFile.ts

```diff
@@ -45,14 +47,67 @@ export async function openYAMLConfigFile(context: IActionContext, node?: StaticW
+    const buildConfigRegex: RegExp = new RegExp(`${buildConfigToSelect}:`, 'g');
+    const buildConfigMatches: RegExpMatchArray | null = configDocumentText.match(buildConfigRegex);
 
-    let offset: number = configDocument.getText().search(configRegex);
-    // Shift the offset to the beginning of the build config's value
-    offset += `${buildConfigToSelect}: `.length;
+    if (buildConfigMatches && buildConfigMatches.length > 1) {
+        void ext.ui.showWarningMessage(localize('foundMultipleBuildConfigs', 'Multiple "{0}" build configurations were found in "{1}".', buildConfigToSelect, basename(configDocument.uri.fsPath)));
+        return undefined;
+    }
+
+    const parsedYaml: Document.Parsed = parseDocument(configDocumentText);
+    type YamlNode = YAMLMap | YAMLSeq | Pair | undefined;
+    const yamlNodes: YamlNode[] = [];
+    let yamlNode: YamlNode = parsedYaml.get('jobs');
+
+    while (yamlNode) {
+        if (yamlNode.hasOwnProperty('key') && yamlNode['key'].value === buildConfigToSelect && yamlNode.hasOwnProperty('value')) {
+            const configValue = <Scalar>yamlNode['value'];
+            const range = <number[] | undefined>configValue.range;
+
+            if (range) {
+                const buildConfigOffset: number = configDocumentText.search(buildConfigRegex);
+                let newlines: number = 0;
+
+                // The range returned from the `yaml` package doesn't include newlines
+                // So count newlines and include them in the range we return
+                for (const char of configDocumentText.slice(0, buildConfigOffset)) {
+                    newlines += char === '\n' ? 1 : 0;
+                }
+
+                const startOffset = range[0] + newlines;
+                let endOffset = range[1] + newlines;
+
+                if (configValue.comment) {
+                    // `endOffset` by default includes the length of the comment
+                    endOffset -= configValue.comment.length + 1;
+
+                    // Don't include the comment character
+                    if (configDocumentText[endOffset] === '#') {
+                        endOffset--;
+                    }
+
+                    // Don't include any horizontal whitespace between the end of the YAML value and the comment
+                    while (/[ \t]/.test(configDocumentText[endOffset - 1])) {
+                        endOffset--;
+                    }
+                }
+
+                const startPosition = configDocument.positionAt(startOffset);
+                const endPosition = configDocument.positionAt(endOffset);
+                return new Range(startPosition, endPosition);
+            }
+        } else if (yamlNode.hasOwnProperty('items')) {
+            yamlNodes.push(...yamlNode['items'])
+        } else if (yamlNode.hasOwnProperty('value') && yamlNode['value'].hasOwnProperty('items')) {
+            yamlNodes.push(...yamlNode['value']['items'])
+        }
```

Comment:
Fixing lint errors and seems easier to pop `yamlNode.value` onto the queue rather than dig into `items` in multiple places

```suggestion
        } else if ('items' in yamlNode) {
            yamlNodes.push(...yamlNode.items);
        } else if ('value' in yamlNode && typeof yamlNode.value === 'object') {
            yamlNodes.push(yamlNode.value)
        }
```